kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/jms/KieServerXAConnectionFactory.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+package org.kie.server.integrationtests.jbpm.jms;
+
+import javax.jms.ConnectionFactory;
+import javax.jms.JMSException;
+import javax.jms.XAConnection;
+import javax.jms.XAConnectionFactory;
+
+import org.kie.server.integrationtests.config.TestConfig;
+
+import java.lang.reflect.Method;
+
+public class KieServerXAConnectionFactory implements XAConnectionFactory {
+
+    public static ConnectionFactory connectionFactory;
+
+    @Override
+    public XAConnection createXAConnection() throws JMSException {
+        return createXAConnection(TestConfig.getUsername(), TestConfig.getPassword());
+    }
+
+    @Override
+    public XAConnection createXAConnection(String userName, String password) throws JMSException {
+        try {
+            Method method = connectionFactory.getClass().getMethod("createXAConnection", String.class, String.class);
+            return (XAConnection) method.invoke(connectionFactory, userName, password);
```

Comment:
would recommend to try to look up remote JTA aware connection factory but if that becomes too difficult I'd say it does not bring much value to be tested on other containers than EAPI/wildfly. or at least not at the moment - mainly because this is standard JMS api and thus was tested multiple times on other places.